web/src/components/zonemap.jsx

```diff
@@ -126,34 +137,21 @@ const ZoneMap = ({
     }
   }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
-  // TODO: Consider moving the calculation to a useMemo function
-  // change color of zones if timeslider is changed
-  useEffect(() => {
-    if (isLoaded && co2ColorScale) {
-      // TODO: This will only change RENDERED zones, so if you change the time in Europe and zoom out, go to US, it will not be updated!
-      // TODO: Consider using isdragging or similar to update this when new zones are rendered
-      const features = ref.current.queryRenderedFeatures();
+  useMemo(() => {
+    if (isLoaded) {
       const map = ref.current.getMap();
-      features.forEach((feature) => {
-        const { color, zoneId } = feature.properties;
-        let fillColor = color;
-        const zoneData = zones[zoneId]?.[selectedTimeAggregate].overviews[selectedZoneTimeIndex];
-
+      zoneValues.forEach((zone, i) => {
+        const zoneData = zone[selectedTimeAggregate].overviews[selectedZoneTimeIndex];
         const co2intensity = zoneData ? getCO2IntensityByMode(zoneData, electricityMixMode) : null;
+        const fillColor = co2ColorScale(co2intensity);
 
-        // Calculate new color if zonetime is selected and we have a co2intensity
-        if (selectedZoneTimeIndex !== null && co2intensity) {
-          fillColor = co2ColorScale(co2intensity);
-        }
-        const existingColor = feature.id
-          ? map.getFeatureState({ source: 'zones-clickable', id: feature.id }, 'color')?.color
-          : color;
+        const existingColor = map.getFeatureState({ source: 'zones-clickable', id: i }, 'color')?.color;
 
-        if (feature.id && fillColor !== existingColor) {
+        if (fillColor !== existingColor) {
```

Comment:
This might not even be required. If we omit this, then we don't need to call `map.getFeatureState` L148. Didn't test which version was fastest.